BusinessCentral.LinterCop/Design/Rule0076TableRelationTooLong.cs

```diff
@@ -0,0 +1,110 @@
+            is not TableRelationPropertyValueSyntax tableRelation)
+            return;
+
+        AnalyzeTableRelations(context, currentField, tableRelation);
+    }
+
+    private void AnalyzeTableRelations(SymbolAnalysisContext context, IFieldSymbol currentField, TableRelationPropertyValueSyntax? tableRelation)
+    {
+        while (tableRelation is not null)
+        {
+            if (tableRelation.RelatedTableField is QualifiedNameSyntax relatedField)
+            {
+                var relatedFieldSymbol = GetRelatedFieldSymbol(
+                    relatedField.Left as IdentifierNameSyntax,
+                    relatedField.Right as IdentifierNameSyntax,
+                    context.Compilation);
+
+                if (relatedFieldSymbol is not null && ShouldReportDiagnostic(currentField, relatedFieldSymbol))
+                {
+                    ReportLengthMismatch(context, currentField, relatedFieldSymbol, relatedField);
+                }
+            }
+
+            tableRelation = tableRelation.ElseExpression?.ElseTableRelationCondition;
+        }
+    }
+
+    private static bool ShouldReportDiagnostic(IFieldSymbol currentField, IFieldSymbol? relatedField) =>
+        relatedField?.HasLength == true &&
+        currentField.HasLength &&
+        currentField.Length < relatedField.Length;
+
+    private static void ReportLengthMismatch(SymbolAnalysisContext context, IFieldSymbol currentField,
+        IFieldSymbol relatedField, QualifiedNameSyntax relatedFieldSyntax)
+    {
+        context.ReportDiagnostic(Diagnostic.Create(
+            DiagnosticDescriptors.Rule0076TableRelationTooLong,
+            currentField.GetLocation(),
+            relatedField.Length,
+            relatedFieldSyntax.ToString(),
+            currentField.Length,
+            currentField.Name));
+    }
+
+    private IFieldSymbol? GetRelatedFieldSymbol(IdentifierNameSyntax? table, IdentifierNameSyntax? field, Compilation compilation)
+    {
+        if (table?.GetIdentifierOrLiteralValue() is not string tableName ||
+            field?.GetIdentifierOrLiteralValue() is not string fieldName)
+            return null;
+
+        return GetFieldFromTable(tableName, fieldName, compilation) ??
+               GetFieldFromTableExtension(tableName, fieldName, compilation);
+    }
+
+    private static IFieldSymbol? GetFieldFromTable(string tableName, string fieldName, Compilation compilation)
+    {
+        var tables = compilation.GetApplicationObjectTypeSymbolsByNameAcrossModules(SymbolKind.Table, tableName);
+        return tables.FirstOrDefault() is ITableTypeSymbol tableSymbol
+            ? tableSymbol.Fields.FirstOrDefault(x => x.Name == fieldName)
+            : null;
+    }
+
+    private static IFieldSymbol? GetFieldFromTableExtension(string tableName, string fieldName, Compilation compilation)
+    {
+        var extensions = compilation.GetDeclaredApplicationObjectSymbols()
```

Comment:
We could improve performance to iterate once to the collection and use the `OfType` instead of a where/cast.

I think you can make this into a single LINQ query like this:

```C#
return compilation.GetDeclaredApplicationObjectSymbols()
    .OfType<ITableExtensionTypeSymbol>()
    .Where(ext => ext.Target?.Name == tableName)
    .SelectMany(ext => ext.AddedFields)
    .FirstOrDefault(field => field.Name == fieldName);

Reply:
Yup, much better. Fixed